plugins/flytekit-ray/flytekitplugins/ray/models.py

```diff
@@ -14,14 +15,22 @@ def __init__(
         min_replicas: typing.Optional[int] = None,
         max_replicas: typing.Optional[int] = None,
         ray_start_params: typing.Optional[typing.Dict[str, str]] = None,
-        k8s_pod: typing.Optional[K8sPod] = None,
+        requests: typing.Optional[Resources] = None,
+        limits: typing.Optional[Resources] = None,
     ):
         self._group_name = group_name
         self._replicas = replicas
         self._max_replicas = max(replicas, max_replicas) if max_replicas is not None else replicas
         self._min_replicas = min(replicas, min_replicas) if min_replicas is not None else replicas
         self._ray_start_params = ray_start_params
-        self._k8s_pod = k8s_pod
```

Comment:
We should keep this as part of the interface and build helper functions that construct valid pod specs instead (as [mentioned](https://github.com/flyteorg/flyte/pull/5933#discussion_r1834548991) in the original flyte PR). This is going to help in the other problem we're having with passing the gpu resource name around (in other words, gpu can be an argument of one of the helper function that builds pod specs).

Reply:
I get what you are saying. So we want users to construct the pod specs themself like calling `construct_k8s_pod_spec_from_resources()` or specifying pod templates in user code?

Reply:
I would make the method name simple, maybe pod from resources